v8env/bridge.js

```diff
@@ -0,0 +1,104 @@
+import { logger } from './logger'
+
+const DEFAULT_BRIDGE_TRANSFER_OPTIONS = {
+  release: true,
+  transfer: false
+}
+
+export default function initBridge(ivm, dispatch) {
+  const bridge = global.bridge = {
+    prepareValue(arg, opts = {}) {
+      if (!arg) // false, undefined, null, 0, "" (all transferable)
+        return arg
+
+      let ctor;
+      // this can bomb with certain values
+      try {
+        ctor = arg.constructor
+      } catch (e) {
+        ctor = undefined
+      }
+
+      switch (ctor) {
+        // transferable ivm
+        case ivm.ExternalCopy:
+        case ivm.Reference:
+
+        // primitives
+        case String:
+        case Number:
+        case Boolean:
+          return arg
+
+        // typed arrays
+        case ArrayBuffer:
+        case Uint8Array:
+        case Int8Array:
+        case Uint8ClampedArray:
+        case Int16Array:
+        case Uint16Array:
+        case Int32Array:
+        case Uint32Array:
+        case Float32Array:
+        case Float64Array:
+          return bridge.wrapValue(arg, opts)
+
+        // transferable if wrapped
+        case DataView:
+        case Map:
+        case Set:
+        case RegExp:
+        case Date:
+        case Object: // plain object
+        case Array:
+          return bridge.wrapValue(arg, { release: !!opts.release, transfer: false })
+
+        case Function:
+          return bridge.wrapFunction(arg)
+
+        // simplified-values
+        case Error:
+          return arg.stack || arg.message || arg.toString()
+
+        default:
+          logger.warn("Unknown constructor, wrapping in ivm.Reference:", ctor.name)
+          return new ivm.Reference(arg);
+        // throw new Error(`Can't prepare a non-transferable value (constructor: '${ctor && ctor.name || 'unknown'}')`);
+      }
+    },
+    dispatch(name, ...args) {
+      logger.debug("dispatch", name)
+      return dispatch.apply(null, [name, ...args.map((a) => bridge.prepareValue(a))])
```

Comment:
This effectively makes `dispatch` accessible. Is there any reason we wouldn't want it to be?

Reply:
I was thinking about that, I think it's fine. Just got to be careful with how we handle it. I think the worse case is we would throw an error. We check the function name passed against a map and are pretty strict about it.